hugie/endpoint.py

```diff
@@ -37,7 +37,33 @@ def list(
     """
     List all the deployed endpoints
     """
-    response = requests.get(settings.endpoint_url, headers=headers)
+
+    try:
+        response = requests.get(
+            f"{settings.endpoint_url}", headers=headers, json={})
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        typer.secho(API_ERROR_MESSAGE, fg=typer.colors.RED)
+        raise SystemExit(e)
+    
+    if response.status_code == 401:
```

Comment:
the flow will never reach here because of `response.raise_for_status` see https://requests.readthedocs.io/en/latest/user/quickstart/?highlight=raise_for_status#response-status-codes

you can move this logic inside the except and before the `raise SystemExit(e)` and that will work.

can you also please add a test for one of these conditions so we know that the flow reaches that point. you can mock the calls to aws.